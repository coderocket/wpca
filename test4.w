x, y,z : int
	{x + 3 >= y }
; x := x + 3
	{ x >= y }

