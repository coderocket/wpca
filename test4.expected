open misc
open util/integer

pred true { no none }
pred false { some none }

one sig State {
 x,y,z : Int
}

one sig Const {
 
}

pred obligation {
(State.x.add[3] >= State.y) => {
State.x.add[3] >= State.y
}
}

check { obligation }
